### ISSUE TITLE
DeserializationFeature.ACCEPT_EMPTY_ARRAY_AS_NULL_OBJECT not respected?

### ISSUE THREAD

**Issue by JPMoresmau** [FasterXML/jackson-databind#1571]
On 2.8.7, I turn on `DeserializationFeature.UNWRAP_SINGLE_VALUE_ARRAYS` and `DeserializationFeature.ACCEPT_EMPTY_ARRAY_AS_NULL_OBJECT`. The JSON:
```
{"id":[]}
```
Cannot deserialize to an instance of the class:
```
	public static class TestArrayBean{
		private String id;
		
		public String getId() {
			return id;
		}
		
		public void setId(String id) {
			this.id = id;
		}
}
```
(Can not deserialize instance of java.lang.String out of END_ARRAY token).
It works if there is a value.
I was hoping to get a non null bean with a null id member.

**Comment by JPMoresmau** [comment]
Duplicate of #994 I think?

**Comment by cowtowncoder** [comment]
@JPMoresmau Yes may well be the same. You may want to close this one, unless you think they differ.

I would hope to address it, but looking at everything I've tagged with `2.9`, a PR would certainly improve chances of this getting fixed for 2.9.0 (and if simple enough, backported in 2.8 too).

**Comment by JPMoresmau** [comment]
Closed as duplicate of #994

**Comment by cowtowncoder** [comment]
FWIW, worked on #994, with (what I hope is) full fix for 2.9.0, partial for 2.8(.8).
